erpnext/manufacturing/doctype/work_order/work_order_list.js

```diff
@@ -17,3 +17,40 @@ frappe.listview_settings['Work Order'] = {
 		}
 	}
 };
+
+frappe.ui.form.on("Work Order", {
```

Comment:
Suggested Change: you directly add this indicator color and condition in work_order_list.js